todo_app/data/trello_client.py

```diff
@@ -0,0 +1,92 @@
+import os
+import requests
+
+TRELLO_KEY = os.environ.get('TRELLO_API_KEY', '')
+TRELLO_TOKEN = os.environ.get('TRELLO_API_TOKEN', '')
+TRELLO_TO_DO_BOARD_ID = os.environ.get('TRELLO_TO_DO_BOARD_ID', '')
```

Comment:
This works nicely here, but pre-loading these may bite you in the next module so worth keeping an eye out for it. 

The reason it could be a problem is that these lines of code run when this file is first _imported_. That works perfectly right now, but:
* When we introduce tests we'll sometimes want to do some setup before we start the app
* This can include loading fake/test environment variables
* But these variables will already have had their values assigned by the time we get to our setup, so we won't affect them

We can fix that by messing with our import order carefully, but we typically end up with very fragile code, so a better solution is one of:
* The easiest here, just load the environment variables when they're needed, e.g. convert `card_request_params` into a function that can be called when required:
```python
def get_card_request_params():
    return {
        'key': os.environ.get('TRELLO_API_KEY', ''),
        'token': os.environ.get('TRELLO_API_TOKEN', ''),
        'cards': 'open',
        'card_fields': 'id,name,desc,due',
    }
```
* Personally I'd be lazy and go with the above here, but we could go with a more classic OOP solution; instantiate the TrelloClient class (i.e. use an object rather than class methods everywhere), and load the environment variables in a constructor, e.g. something like:
```python
class TrelloClient:

    def __init__(self):
        self.TRELLO_KEY = os.getenv('TRELLO_API_KEY')
        ...

    def get_trello_lists(self):
        """
        Fetches the list of all lists available in the Trello
        Returns:
            The json response from the Trello API (the list of Trello lists if successful)
        """
        return requests.get(self.LISTS_URL, params=self.card_request_params).json()
```

Reply:
Thanks for this. It feels off to need to change how we load the environment variables since by definition they'll always be constant when the app is actually running (and if for some reason they were changed on the server it would feel like non-standard behaviour for an already running app to pick up those changes). 

As it happens I'd already run into this on exercise 3, but I just used the solution I've seen used in projects elsewhere and moved the environment variable overrides into conftest.py